### PR TITLE
chore: improve error handler

### LINKED ISSUE #242: Put an HTML error handler
As a user, I can see a beautiful error page.

### Motivation

JSON is not very user friendly you know...

### Detail

Very easy to do it using [express-err](https://github.com/neoziro/express-err/).

### PR DESCRIPTION
**Pull request by neoziro**
Closes #242 

screen shot 2017-04-01 at 18 19 51

Message and stack trace are not displayed in production.

### CODE REVIEW

File: src/review/routes/Error.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import React, { PropTypes } from 'react'
+import { compose } from 'recompact'
+import { connect } from 'react-redux'
+import { withStyles, createStyleSheet } from 'material-ui/styles'
+import Paper from 'material-ui/Paper'
+import ViewContainer from 'modules/components/ViewContainer'
+import ScrollView from 'modules/components/ScrollView'
+import ReviewAppBar from 'review/modules/AppBar/AppBar'
+
+const styleSheet = createStyleSheet('ProductHome', () => ({
```
Comment (oliviertassinari) [inline src/review/routes/Error.js]: `'Error'`

File: src/review/routes/Error.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  landing: {
+    padding: 15,
+  },
+}))
+
+const Error = ({ classes, error }) => (
+  <ViewContainer>
+    <ReviewAppBar />
+    <ScrollView>
+      <Paper rounded={false} elevation={3} className={classes.landing}>
```
Comment (oliviertassinari) [inline src/review/routes/Error.js]: Why not using the `LayoutBody`?

File: src/review/routes/Error.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        { <div style={{ fontSize: '30px', lineHeight: 2 }}>Error {error.statusCode}</div> }
```
Comment (oliviertassinari) [inline src/review/routes/Error.js]: `fontSize: 30,`

Comment (oliviertassinari) [inline src/review/routes/Error.js]: CSS?

File: src/review/routes/Error.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  error: PropTypes.shape({
+    message: PropTypes.string,
+    stack: PropTypes.string,
+    statusCode: PropTypes.number.isRequired,
+  }),
+}
+
+export default compose(
+  withStyles(styleSheet),
+  connect(({ data: { error } }) => ({ error })),
```
Comment (oliviertassinari) [inline src/review/routes/Error.js]:
```js
connect((state) => ({ error: state.data.error })),
```
?
Looks much better for grepping

File: src/review/routes/Error.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+import ScrollView from 'modules/components/ScrollView'
+import ReviewAppBar from 'review/modules/AppBar/AppBar'
+
+const styleSheet = createStyleSheet('ProductHome', () => ({
+  landing: {
+    padding: 15,
+  },
+}))
+
+const Error = ({ classes, error }) => (
```
Comment (oliviertassinari) [inline src/review/routes/Error.js]: function to be consistent

File: src/review/routes/homepage/Homepage.js [hunk 6]
```diff
@@ -1,13 +1,23 @@
 import React, { PropTypes } from 'react'
 import { connect } from 'react-redux'
 import ProductHome from 'review/routes/product/Home'
+import Error from 'review/routes/Error'
 import Dashboard from 'review/routes/dashboard/Dashboard'
 
-function Homepage(props) {
-  return props.user ? <Dashboard /> : <ProductHome />
+function Homepage({ user, error }) {
```
Comment (oliviertassinari) [inline src/review/routes/homepage/Homepage.js]: I would rather use `props` to be consistent.

File: src/server/middlewares/rendering.js [hunk 7]
```diff
@@ -52,7 +52,7 @@ function injectJSON(data) {
   return JSON.stringify(data, null, process.env.NODE_ENV === 'production' ? 0 : 2)
 }
 
-export default (req, res) => {
+export default additionalClientData => (req, res) => {
```
Comment (oliviertassinari) [inline src/server/middlewares/rendering.js]: What about moving `additionalClientData` into the `response`?

File: src/review/routes/homepage/Homepage.js (hunk 6 above)
Comment (oliviertassinari) [inline src/review/routes/homepage/Homepage.js]: Actually, this pattern is breaking https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prop-types.md.

File: src/review/routes/Error.js (hunk 1 above)
Comment (neoziro) [inline src/review/routes/Error.js]: Ok

File: src/review/routes/Error.js (hunk 5 above)
Comment (neoziro) [inline src/review/routes/Error.js]: Less code is better?

File: src/review/routes/Error.js (hunk 2 above)
Comment (neoziro) [inline src/review/routes/Error.js]: Yes I should use it.

File: src/review/routes/Error.js (hunk 3 above)
Comment (neoziro) [inline src/review/routes/Error.js]: Yes, but not a problem for now.

File: src/review/routes/Error.js (hunk 4 above)
Comment (neoziro) [inline src/review/routes/Error.js]: What grepping?

File: src/review/routes/homepage/Homepage.js (hunk 6 above)
Comment (neoziro) [inline src/review/routes/homepage/Homepage.js]: No it doesn't break it I had the warning. I think destructuring is more explicit and dryer.

File: src/server/middlewares/rendering.js (hunk 7 above)
Comment (neoziro) [inline src/server/middlewares/rendering.js]: I don't understand, if it's mutating response, I think it's not a good idea.

File: src/review/routes/Error.js (hunk 4 above)
Comment (neoziro) [inline src/review/routes/Error.js]: You should grep "connect(".

File: src/review/routes/homepage/Homepage.js (hunk 6 above)
Comment (oliviertassinari) [inline src/review/routes/homepage/Homepage.js]: Right, it's working.

File: src/review/routes/Error.js (hunk 4 above)
Comment (oliviertassinari) [inline src/review/routes/Error.js]: I don't know, it feels more idiomatic without the destructuration.

Comment (oliviertassinari) [inline src/review/routes/Error.js]: by grepping, I'm referring to looking for where the `state.data` is used.

File: src/review/routes/Error.js (hunk 5 above)
Comment (oliviertassinari) [inline src/review/routes/Error.js]: From my point of view, flexibility and consistency are serious to less code.

File: src/server/middlewares/rendering.js (hunk 7 above)
Comment (oliviertassinari) [inline src/server/middlewares/rendering.js]: Don't we already do that with `req.user`? We could mutate the request otherwise 🙈

Comment (neoziro) [inline src/server/middlewares/rendering.js]: Passport do that yes, but I don't agree with it.

**Pull request merged**